test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/CheckAllAnswersAndSubmitControllerSpec.scala

```diff
@@ -1310,7 +1313,12 @@ class CheckAllAnswersAndSubmitControllerSpec
           inSequence {
             mockAuthWithNoRetrievals()
             mockGetSession(sessionWithJourney(completeFillingOutReturnWithRepresenteeWithNoReference))
-            mockCgtRegistrationService(mockRepresenteeCgtReference)
+            mockCgtRegistrationService(
+              Right(mockRepresenteeCgtReference),
+              mockedCompleteReturn.representeeAnswers
```

Comment:
it'd be better to factor out the `representeeAnswers` that are generated in a separate `val` so that you can reference it here instead of handling the case `None` when it can never happen. A similar thing happens in the test below